src/Perla.Plugins/Library.fsi

```diff
@@ -0,0 +1,158 @@
+namespace Perla.Plugins
+
+open System.Threading.Tasks
+
+/// This type handles the file extensions for bessed types
+/// like HTML, CSS, JS, any other extension is considered a
+/// "Custom" extension
+[<RequireQualifiedAccess; Struct>]
+type FileExtension =
+    | JS
+    | CSS
+    | HTML
+    | Custom of string
+
+type FileTransform =
+    {
+        /// The path where the file changed event was activated
+        /// This value should not be modified between plugin transformations
+        /// because it could break further processing of the sfile
+        originalPath: string
```

Comment:
If you cannot update this field, I would suggest making a `FileTransformRequest` and `FileTransformResponse` type approach to it.

Reply:
The idea of these plugins is that they will run at build time and when the dev server starts, a file changes, so I don't necessarily want to introduce something that ties to dev/build time specifically I'd like to have it agnostic

The plugin should be able to run without caring who is calling it at what time, since I will have control when these are called I might as well ensure that the originalPath stays as is even if users modify this value

But! that being said I'd like to know what do you have in mind for that, could you elaborate a little bit more on this please?

Reply:
Just something like:
```fsharp
type FileTransformRequest =
   { OriginalPath: string
     Content: string
     Extension: string }

type FileTransformResponse =
  { Content: string
    Extension: string }
```

Given that you mentioned `originalPath` should not be updated.